src/app/OperationalSessionSetup.cpp

```diff
@@ -305,9 +307,14 @@ CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessagePro
     mCASEClient = mClientPool->Allocate();
     ReturnErrorCodeIf(mCASEClient == nullptr, CHIP_ERROR_NO_MEMORY);
 
+    // Session has been established. This is when discovery is also stopped
```

Comment:
I don't get this comment.  The session has not been established at this point....

And as we discussed discovery is sill ongoing, in parallel with the CASE bits.  So I'm a little confused by this part.